src/entities/Cinema.java

```diff
@@ -10,12 +10,12 @@ public class Cinema implements Serializable {
     private HashMap<String, ArrayList<LocalDate>> showtimes;
     private HashMap<String, ArrayList<Seat>> seats;  //key is the movie title + showtime.toString().
     private String cinemaName;
-    public Cinema(String cinemaName, ArrayList<Movie> movies, HashMap<Movie, ArrayList<LocalDate>> showtimes, HashMap<String, ArrayList<Seat>> seats) {
+    public Cinema(String cinemaName, ArrayList<Movie> movies, HashMap<String, ArrayList<LocalDate>> showtimes, HashMap<String, ArrayList<Seat>> seats) {
         this.movies = movies;
         this.cinemaName = cinemaName;
-        ArrayList<LocalDate> shows = new ArrayList<>();
+        ArrayList<LocalDate> showstime = new ArrayList<>();
         this.showtimes = new HashMap<>();
-        for (Movie mov : showtimes.keySet()) {
+        for (Movie mov : movies) {
```

Comment:
This one i'm not too sure what showtimes.keySet() are but changing it to movies fixed the error it had.

Reply:
keySet() returns an iterator for the set of keys in the HashMap. It was throwing an error becasue the key in the hashmap is a string of movie title, and not a Movie object, so that was definitely an error.